pkg/cmd/get/clusters.go

```diff
@@ -21,14 +67,313 @@ func preClustersE(cmd *cobra.Command, args []string) error {
+
+			// Print the internal port running the Kuber API service
+			kubeApiPort, err := findInternalKubeApiPort(cli)
+			if err != nil {
+				//logger.Error(err, "failed to get the kubernetes default service.")
+				return nil, err
+			} else {
+				aCluster.KubePort = kubeApiPort
+			}
+
+			// Let's check what the current node reports
+			var nodeList corev1.NodeList
+			err = cli.List(context.TODO(), &nodeList)
+			if err != nil {
+				//logger.Error(err, "failed to list nodes for the current kube cluster.")
+				return nil, err
+			}
+
+			for _, node := range nodeList.Items {
+				nodeName := node.Name
+
+				aNode := Node{}
+				aNode.Name = nodeName
+
+				for _, addr := range node.Status.Addresses {
+					switch addr.Type {
+					case corev1.NodeInternalIP:
+						aNode.InternalIP = addr.Address
+					case corev1.NodeExternalIP:
+						aNode.ExternalIP = addr.Address
+					}
+				}
+
+				// Get Node capacity
+				aNode.Capacity = populateNodeCapacity(node.Status.Capacity)
+
+				// Get Node Allocated resources
+				allocated, err := printAllocatedResources(context.Background(), cli, node.Name)
+				if err != nil {
+					//logger.Error(err, "failed to get the allocated resources.")
+					return nil, err
+				}
+				aNode.Allocated = allocated
+			}
+		}
+		clusterList = append(clusterList, aCluster)
+	}
+
+	return clusterList, nil
+}
+
+func generateClusterTable(clusterTable []Cluster) metav1.Table {
+	table := &metav1.Table{}
+	table.ColumnDefinitions = []metav1.TableColumnDefinition{
+		{Name: "Name", Type: "string"},
+		{Name: "External-Port", Type: "string"},
+		{Name: "Kube-Api", Type: "string"},
+		{Name: "TLS", Type: "string"},
+		{Name: "Kube-Port", Type: "string"},
+	}
+	for _, cluster := range clusterTable {
+		row := metav1.TableRow{
+			Cells: []interface{}{
+				cluster.Name,
+				cluster.ExternalPort,
+				cluster.URLKubeApi,
+				cluster.TlsCheck,
+				cluster.KubePort,
+			},
+		}
+		table.Rows = append(table.Rows, row)
 	}
+	return *table
+}
+
+func printTable(opts printers.PrintOptions, table metav1.Table) {
+	logger := helpers.CmdLogger
+	out := bytes.NewBuffer([]byte{})
+	printer := printers.NewTablePrinter(opts)
+	err := printer.PrintObj(&table, out)
+	if err != nil {
+		logger.Error(err, "failed to print the table.")
+		return
+	}
+	fmt.Println(out.String())
+}
+
+func populateNodeCapacity(resources corev1.ResourceList) Capacity {
+	capacity := Capacity{}
+	for name, quantity := range resources {
+		if strings.HasPrefix(string(name), "hugepages-") {
+			continue
+		}
+
+		if name == corev1.ResourceMemory {
+			memoryInBytes := quantity.Value()                           // .Value() gives the value in bytes
+			memoryInGB := float64(memoryInBytes) / (1024 * 1024 * 1024) // Convert to GB
+			capacity.Memory = memoryInGB
+		}
+
+		if name == corev1.ResourceCPU {
+			capacity.Cpu = quantity.Value()
+		}
+
+		if name == corev1.ResourcePods {
+			capacity.Pods = quantity.Value()
+		}
+
+	}
```

Comment:
Since we know the names of resources we want, we can look it up in map directly. e.g.

```go
cap, ok := resources[corev1.ResourceMemory]
```

Reply:
Fixed. See: 68bc257